### PR TITLE
Challenges 70, 71 and 72 completed.

### PR DESCRIPTION
**Pull request by gaebar**

### CODE REVIEW

File: exercises/tuples-lists-dictionaries/challenge_072.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+"""
+Create a list of six school subjects. Ask the user which of these subjects they don't like. 
+Delete the subject they have chosen from the list before you display the list again.
+"""
+
+subjects_list = ["Maths", "Literature", "Science", "History", "Spanish", "Computing"]
+print(subjects_list)
+dislike = input("Which of these subjects do you dislike? ")
+erase_from_list = subjects_list.index(dislike)
+del subjects_list[erase_from_list]
```
Comment (rrees) [inline exercises/tuples-lists-dictionaries/challenge_072.py]: What happens here when the user types something that wasn't in the list?

File: exercises/tuples-lists-dictionaries/challenge_070.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+country_names = ("England", "Italy", "Spain", "France", "Portugal")
+
+print(country_names)
+
+
+# Ask for user input and ensures the user enters an integer
+def get_user_input_digit(message):
+    while True:
+        user_input = input(message)
+        if not user_input.isdigit():
```
Comment (rrees) [inline exercises/tuples-lists-dictionaries/challenge_070.py:30]: What happens here if there are more than 10 countries in the list and a user wants to display them

Comment (gaebar) [reply exercises/tuples-lists-dictionaries/challenge_070.py:30]: I have added a parameter for the max index and added a join to convert the tuple to string

File: exercises/tuples-lists-dictionaries/challenge_072.py (hunk 1 above)
Comment (gaebar) [reply exercises/tuples-lists-dictionaries/challenge_072.py]: Corrected to add a check to ensure the user choice is in the list

File: exercises/tuples-lists-dictionaries/challenge_072.py [hunk 3]
```diff
@@ -6,9 +6,27 @@
...
+
+print(f"Subjects list: {join_subjects(subjects_list)}")
+
+
+def ask_user_input():
+    while True:
+        user_input = input("Which one of these subjects do you dislike? ")
+        if user_input in subjects_list:
+            erase_from_list = subjects_list.index(user_input)
+            del subjects_list[erase_from_list]
```
Comment (rrees) [inline exercises/tuples-lists-dictionaries/challenge_072.py]: There is still a problem here if the user enters invalid input. Try entering an invalid input and seeing what happens.

File: exercises/tuples-lists-dictionaries/challenge_070.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            print("Oops! That was not a valid number. Try again...")
+        return int(user_input)
+
+
+def ask_for_country_by_index():
+    max_index = len(country_names) - 1
```
Comment (rrees) [inline exercises/tuples-lists-dictionaries/challenge_070.py:37]: What happens here if there are more than 10 country names though, does this code still work?

File: exercises/tuples-lists-dictionaries/challenge_072.py (hunk 3 above)
Comment (gaebar) [reply exercises/tuples-lists-dictionaries/challenge_072.py]: I have updated the challenge to include more subjects and also I am checking for lowercase inputs. Everything seems to be working now.

File: exercises/tuples-lists-dictionaries/challenge_070.py (hunk 4 above)
Comment (gaebar) [reply exercises/tuples-lists-dictionaries/challenge_070.py:37]: Added more countries and restored the else statement

File: exercises/tuples-lists-dictionaries/challenge_072.py [hunk 5]
```diff
@@ -5,10 +5,47 @@
...
+def ask_user_input():
+    while True:
+        user_input = input("Which one of these subjects do you dislike? ")
+        user_input_lowercase = user_input.lower()
+
+        if user_input_lowercase in subjects_list_lowercase:
+            index_to_delete = subjects_list_lowercase.index(user_input_lowercase)
+            subject_to_delete = subjects_list[index_to_delete]
+
+            del subjects_list[index_to_delete]
```
Comment (rrees) [inline exercises/tuples-lists-dictionaries/challenge_072.py:41]: Can you think of a way to do this without the double delete action (one of which must be redundant).

Comment (gaebar) [reply exercises/tuples-lists-dictionaries/challenge_072.py:41]: I have removed the deletion of the lowercase list items, as it was not used.

**Review by rrees — APPROVED** [review]

**Pull request merged**